### PR TITLE
[CS-3088] Font weight work-around

### PR DESCRIPTION
**Pull request by oidoug**
### Description

This PR adds supported font-weight variations for font OpenSans-Regular on android.

Differently than on iOS, on Android, the only font used after Theme translation by `restyle` is `OpenSans-Regular`. This can be easily tested by trying to use the wrong font-weight on any Text, it will show an error with the available options for the font.

- [x] Completes #(CS-3088)

### Checklist

- [x] All UI changes have been tested on a small device

### Screenshots

Screenshot

### CODE REVIEW

**Review by danibonilha — COMMENTED** [review]
The change looks good, but I noticed there are places we are using `fontWeight` and other `weight`,  in order to to apply this across the app we would need to replace the `fontWeight` with `weight` in all used Text reestyle components and not allow `Text` to receive `fontWeight` anymore to avoid making a android mistake, an easy way, is to apply this on the `Text.ts` line 13: 

```ts
export type TextProps = Omit<ShopifyTextProps<Theme>, 'fontWeight'> &
```

File: src/components/walletconnect-list/WalletConnectExplainer.js [hunk 1]
```diff
@@ -25,10 +25,10 @@ const IconQRScanImg = styled(ImgixImage).attrs({
 const CustomText = props => (
   <Text
     color="blueText"
-    fontWeight={props.bold ? '500' : '400'}
     letterSpacing={0.32}
     size="xs"
     textAlign="center"
+    weight={props.bold ? 'bold' : 'regular'}
     {...props}
```
Comment (danibonilha) [inline src/components/walletconnect-list/WalletConnectExplainer.js]: nit: This one already passes the props forward, we could remove this condition, and on `   <CustomText bold>` replace with `   <CustomText weight='bold'>`

**Review by danibonilha — COMMENTED** [review]
small suggestion, and missing some eslint fixes, once it's green, it's good to go.

File: src/components/walletconnect-list/WalletConnectExplainer.js (hunk 1 above)
Comment (oidoug) [reply src/components/walletconnect-list/WalletConnectExplainer.js]: Good catch 👍

**Comment by oidoug** [comment]
@danibonilha , I refactored all Texts to use weight and added the Omit as you suggested.

There are a few places that are still using `fontWeight` but they are either for Inputs or are not using the Theme (prob. older Rainbow implementation). If you don't mind I'll leave these cases behind for now.

**Review by danibonilha — APPROVED** [review]
🚀

**Pull request merged**